src/rebootstrap-raft/main.go

```diff
@@ -123,6 +134,52 @@ func (c *rebootstrapCommand) Run(ctx *cmd.Context) error {
 	return errors.Trace(c.bootstrapRaft(raftServers))
 }
 
+func (c *rebootstrapCommand) getStatePassword(machineID string) (string, error) {
+	agentConfigPath := fmt.Sprintf("/var/lib/juju/agents/machine-%s/agent.conf", machineID)
+	file, err := os.Open(agentConfigPath)
+	if err != nil {
+		return "", errors.Annotatef(err, "opening agent config file %q", agentConfigPath)
+	}
+	defer file.Close()
+
+	var config map[string]interface{}
+	decoder := yaml.NewDecoder(file)
```

Comment:
I know the existing codebase doesn't have tests, but it would probably be good to add a couple about finding machine-X in a directory and parsing an agent.conf file to extract its statepassword.
None of this command actually depends on the receiver (c *rebootstrapCommand) so it certainly could be split into a function that just takes a filename (or file), and that could be tested directly.
Taking a file means you could just expect an io.Reader and then pass in a bufio.Reader for the test suite.

Reply:
I pushed a first attempt at unit tests for the state password function.